components/gene-panels/api/src/test/java/org/phenotips/panels/internal/DefaultGenePanelFactoryImplTest.java

```diff
@@ -0,0 +1,289 @@
+        Assert.assertEquals(Collections.emptyList(), genePanel.getTermsForGeneList());
+
+        final JSONObject expectedJson = new JSONObject().put(SIZE_LABEL, 0).put(TOTAL_SIZE_LABEL, 0)
+            .put(GENES_LABEL, new JSONArray());
+        Assert.assertTrue(genePanel.toJSON().similar(expectedJson));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void checkBuildWorksIfFeaturesNotEmpty()
+    {
+        final Set<Feature> features = new HashSet<>();
+        final Feature presentFeature = mock(Feature.class);
+        final Feature absentFeature = mock(Feature.class);
+        final VocabularyTerm presentTerm = mock(VocabularyTerm.class);
+        final VocabularyTerm absentTerm = mock(VocabularyTerm.class);
+        final Object associatedGenes = new ArrayList<>();
+        ((List<String>) associatedGenes).add(GENE1);
+        ((List<String>) associatedGenes).add(GENE2);
+        features.add(presentFeature);
+        features.add(absentFeature);
+        when(presentFeature.isPresent()).thenReturn(Boolean.TRUE);
+        when(absentFeature.isPresent()).thenReturn(Boolean.FALSE);
+        when(presentFeature.getValue()).thenReturn(HPO_TERM1);
+        when(absentFeature.getValue()).thenReturn(HPO_TERM2);
+        when(this.hpo.getTerm(HPO_TERM1)).thenReturn(presentTerm);
+        when(this.hpo.getTerm(HPO_TERM2)).thenReturn(absentTerm);
+        when(presentTerm.get(ASSOCIATED_GENES)).thenReturn(associatedGenes);
+        when(presentTerm.getName()).thenReturn(HPO_TERM1);
+        when(absentTerm.getName()).thenReturn(HPO_TERM2);
+        when(this.hgnc.getTerm(anyString())).thenReturn(null);
+        final GenePanel genePanel = this.genePanelFactory.build(features);
+        // The gene panel should only represent two genes.
+        Assert.assertEquals(2, genePanel.size());
+
+        final Set<VocabularyTerm> presentTerms = new HashSet<>();
+        presentTerms.add(presentTerm);
+        final Set<VocabularyTerm> absentTerms = new HashSet<>();
+        absentTerms.add(absentTerm);
+        // There should only be one present term and one absent term.
+        Assert.assertEquals(presentTerms, genePanel.getPresentTerms());
+        Assert.assertEquals(absentTerms, genePanel.getAbsentTerms());
+
+        final List<TermsForGene> termsForGene = genePanel.getTermsForGeneList();
+        // There should only be two objects, one for each gene.
+        Assert.assertEquals(2, termsForGene.size());
+
+
+        Assert.assertEquals(GENE2, termsForGene.get(0).getGeneId());
+        Assert.assertEquals(GENE2, termsForGene.get(0).getGeneSymbol());
+        // The number of terms associated with "gene1" should be 1.
+        Assert.assertEquals(1, termsForGene.get(0).getCount());
+        // Check the term associated with "gene1" is the one that is expected.
+        Assert.assertEquals(presentTerms, termsForGene.get(0).getTerms());
+
+
+        Assert.assertEquals(GENE1, termsForGene.get(1).getGeneId());
```

Comment:
It's a coincidence, and tests fail on my computer. The result should always have a deterministic sorting.